pkg/cmd/repo/garden/http.go

```diff
@@ -0,0 +1,106 @@
+package garden
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/cli/cli/internal/ghinstance"
+	"github.com/cli/cli/internal/ghrepo"
+)
+
+func getCommits(client *http.Client, repo ghrepo.Interface, maxCommits int) ([]*Commit, error) {
+	type Item struct {
+		Author struct {
+			Login string
+		}
+		Sha string
+	}
+
+	type Result []Item
+
+	commits := []*Commit{}
+
+	pathF := func(page int) string {
+		return fmt.Sprintf("repos/%s/%s/commits?per_page=100&page=%d", repo.RepoOwner(), repo.RepoName(), page)
+	}
+
+	page := 1
+	paginating := true
+	fmt.Println("gathering commits; this could take a minute...")
```

Comment:
Protip: you can now use `opts.IO.StartProgressIndicator()` + `opts.IO.StopProgressIndicator()` to get the spinner rendering 👌